### DISCUSSION TITLE
Add blur to menus and imgui !!!

### DISCUSSION THREAD

**Discussion by PEXCH-SENTXMENT** [danielkrupinski/Osiris#3195] [Ideas]
I think that if you add a blur in the color settings (for the menu), it will greatly increase the attractiveness of the cheat.
Yes, and I myself really want to see it, but I did not understand the imgui and do not know how it works (do not want).
![image](https://user-images.githubusercontent.com/63977167/129482466-d053f838-df9d-4373-9622-7a8de18bd281.png)
I'm not sure if it's blurry, but it looks awesome.
And if you add blur, it's top.

**Comment by linuxer9** [comment]
qo0 doesn't have blur effect, it's just transparency.
for blur effect you can use [GOESP](https://github.com/danielkrupinski/GOESP) or paste the code into your osiris build.

**Reply by PEXCH-SENTXMENT** [reply]
Thnx

**Comment by Frouk3** [comment]
> And if you add blur, it's top.

btw nice english xD

**Reply by PEXCH-SENTXMENT** [reply]
Sry i'm from Russia

**Reply by M3351AN** [reply]
du nut tink i m no a gud man i say poor engleshi thougn

**Reply by diekunstderfuge** [reply]
Talking like that I could swear you were Brazilian. Brazilians frequently use "top" just the way you did.

**Comment by free0ne87** [comment]
Just transparecy, you can probably set it in style menu in custom under different colors. Alpha is transparecy.

**Comment by TTommyGS** [comment]
Yes its a transparency but there is some effect behind it aswell which looks like blur.
I looked trough the src code and didnt find the reason behind this.
Can anyone show an example of how to do that "blur effect" even if its not blur.
Kind Regards

**Comment by free0ne87** [comment]
You can make work this for you, needs slight modification. You can also try to pass only drawlists you needbto blur.



https://github.com/danielkrupinski/GOESP/search?q=Blur


https://drive.google.com/file/d/1minrVa4Nm1kr83ZlTMaAS9G9vfZKJq3_/view?usp=sharing


PostProcessing::performFullscreenBlur(ImGui::GetBackgroundDrawList(), gui->getTransparency());

**Reply by ghost** [reply]
can u make an example please :) for DIRECT X 9

**Reply by ghost** [reply]
how i can use this ???

Shaders::SetDevice(g_pd3dDevice);
Shaders::NewFrame();
Shaders::GausenBlur(ImGui::GetBackgroundDrawList(), ImVec4(0, 0, 0, 0.5));
Shaders::ClearBlur();
Shaders::DeviceReset();

**Reply by ghost** [reply]
i've already fix it
